src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> previous;
+
+        Node(Node<E> previous, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.previous = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = findNode(index);
+        Node<T> previousNode = currentNode.previous;
+        Node<T> newNode = new Node<>(previousNode, value, currentNode);
+        if (previousNode == null) {
+            head = newNode;
+        } else {
+            previousNode.next = newNode;
+        }
+        currentNode.previous = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexExist(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexExist(index);
+        Node<T> currentNode = findNode(index);
+        T previousItem = currentNode.item;
+        currentNode.item = value;
+        return previousItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexExist(index);
+        Node<T> currentNode = findNode(index);
+        T remove = currentNode.item;
+        unlinkNode(currentNode);
+        return remove;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentItem = head;
+        for (int i = 0; i < size; i++) {
```

Comment:
where do you use i in the loop?

Reply:
I don`t use i, so it should be replaced by while loop